src/Features/Core/Portable/ExtractMethod/IExtractMethodService.cs

```diff
@@ -10,6 +10,6 @@ namespace Microsoft.CodeAnalysis.ExtractMethod
 {
     internal interface IExtractMethodService : ILanguageService
     {
-        Task<ExtractMethodResult> ExtractMethodAsync(Document document, TextSpan textSpan, OptionSet options = null, CancellationToken cancellationToken = default);
+        Task<ExtractMethodResult> ExtractMethodAsync(Document document, TextSpan textSpan, bool extractLocalFunction = false, OptionSet options = null, CancellationToken cancellationToken = default);
```

Comment:
would likely be cleaner as exposing two separate functions.  i.e. ExtractMethodAsync and ExtracLocalFunctionAsync.  I don't care hugely though.

This boolean sholud be non-optional though